documentation/modules/ROOT/pages/integration-with-aap.adoc

```diff
@@ -0,0 +1,176 @@
+
+[#aap-components]
+== Defining AAP Components
+
+[#automation-hub]
+=== Automation Hub
+
+The OpenShift AAP operator's Automation Hub, defined by the custom resource definition AutomationHub, is a downstream version of link:https://github.com/ansible/galaxy_ng[Galaxy NG] which allows for the hosting of Ansible collections on a private distribution server. Collections are a distribution format for Ansible objects including playbooks, roles, modules, and plugins. We store all necessary Ansible collections through an Automation Hub deployed on OpenShift, allowing all dependencies to be hosted and managed through our local cluster. We can upload collections, manage collections, and interact with the Automation Hub through API or through its web UI deployed at automation-hub-aap.apps.hub.5g-deployment.lab.
+
+An example AutomationHub CRD is shown below.
+
+[source,bash]
+-----
+apiVersion: automationhub.ansible.com/v1beta1
+kind: AutomationHub
+metadata:
+  name: automation-hub
+  namespace: aap
+spec:
+  route_tls_termination_mechanism: Edge
+  ingress_type: Route
+  loadbalancer_port: 80
+  file_storage_size: 100Gi
+  image_pull_policy: IfNotPresent
+  web:
+    replicas: 1
+  file_storage_access_mode: ReadWriteOnce
+  content:
+    log_level: INFO
+    replicas: 2
+  postgres_storage_requirements:
+    limits:
+      storage: 50Gi
+    requests:
+      storage: 8Gi
+  api:
+    log_level: INFO
+    replicas: 1
+  postgres_resource_requirements:
+    limits:
+      cpu: 1000m
+      memory: 8Gi
+    requests:
+      cpu: 500m
+      memory: 2Gi
+  loadbalancer_protocol: http
+  resource_manager:
+    replicas: 1
+  storage_type: File
+  worker:
+    replicas: 2
+-----
+
+[#automation-controller]
+=== Automation Controller
+
+The Openshift AAP operator's Automation Controller, defined by the custom resource definition AutomationController, deploys an Ansible Automation Controller (formerly Ansible Tower) in our OpenShift cluster. Downstream of the link:https://github.com/ansible/awx[AWX Project] in link:https://github.com/ansible/awx-operator[operator form], our deployed Automation Controller provides a user interface, REST API, and task engine for executing Ansible jobs. The Automation Controller's functionality includes managing Ansible automation objects including projects, inventories, playbooks, roles, credentials, and job templates, as well as the responsibility of invoking and monitoring the execution of Ansible jobs. 
+
+Details of the contents of an Ansible job will be covered as we craft our automation template. At a high level, an Ansible _job_ is a running instance of a _job template_, which contains a logical collection of Ansible _playbooks_ in the form of a _project_, a list of managed hosts in the form of an _inventory_, and any extra variables we want to pass to the job. 
+
+An example AutomationController CRD is shown below.
+
+[source,bash]
```

Comment:
source,yaml

Reply:
After having read the other sections, if we pre-configure the whole AAP instance I don't think this will be relevant for this lab anymore.